### PR TITLE
Upgrade Nginx dependency to patch converge error

### PR DESCRIPTION
**Pull request by zwall-bp**
For CentOS-7, there is an issue with the utilized `12.0.2` version of the `nginx`cookbook, which leads to a package install error, as the `nginx_install` resource attempts to install the nginx package while disabling all repositories, but Nginx's own. `pcre2` is not available when trying to install Nginx.

[`12.1.0`](https://github.com/sous-chefs/nginx/compare/12.0.12...12.1.0#diff-1b04bdb088a472d6c411d783ba1b00874ae8f797db18a7ebef0bc649782651faR99) Patched this by installing `pcre2` before continuing on with the normal install.

This PR is set to install, at the oldest, version `12.2.0`.

### CODE REVIEW

**Comment by ramereth** [comment]
@sectopodwreck can you please fix the CI issues?

**Comment by sectopodwreck** [comment]
> @sectopodwreck can you please fix the CI issues?

Oria patched this issue, along with other conflicting files, in her PR. If you don't mind approving hers first, I can rebase and re-push this branch.

**Comment by ramereth** [comment]
@sectopodwreck I'm seeing failures with tests in the `vhost-centos-7` suite and a converge error in `cookbook-test-centos-7`. Can you please resolve those?

File: test/cookbooks/osl-nginx-test/recipes/default.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 include_recipe 'osl-nginx'
 
+file '/etc/nginx/conf.d/test-kitchen.conf' do
+  content 'server_names_hash_bucket_size 128;'
```
Comment (ramereth) [inline test/cookbooks/osl-nginx-test/recipes/default.rb:4]: This needs to notify restarting the nginx service so the config is active. Can you please add that?

File: test/integration/cookbook-test/inspec/cookbook_test_spec.rb [hunk 2]
```diff
@@ -101,3 +101,7 @@
     its('content') { should match %r{^include /etc/nginx/conf.http.d/#{f}.osuosl.org.conf;$} }
   end
 end
+
+describe file '/etc/nginx/conf.d/test-kitchen.conf' do
+  its('content') { should match "server_names_hash_bucket_size 128" }
+end
```
Comment (ramereth) [inline test/integration/cookbook-test/inspec/cookbook_test_spec.rb]: Please don't add this, it's not needed to test this.

**Comment by ramereth** [comment]
!bump minor ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v6.4.0.  Have a nice day!

**Pull request merged**